test/v2/pid.cpp

```diff
@@ -7,9 +7,38 @@
 
 #include <boost/test/unit_test.hpp>
 
+#include <algorithm>
+#include <vector>
 
 BOOST_AUTO_TEST_CASE(test_pid)
 {
     namespace bp2 = boost::process::v2;
     BOOST_CHECK_NE(bp2::current_pid(), static_cast<bp2::pid_type>(0));
-}
\ No newline at end of file
+
+    auto all = bp2::all_pids();
+    auto itr = std::find(all.begin(), all.end(), bp2::current_pid());
+    BOOST_CHECK(itr != all.end());
+
+    std::vector<bp2::pid_type> children, grand_children; 
+    auto grand_child_pids = [](bp2::pid_type pid, 
+        std::vector<bp2::pid_type> & children, 
+        std::vector<bp2::pid_type> & grand_children) 
+    {  
+        children = bp2::child_pids(pid);
+        for (unsigned i = 0; i < children.size(); i++) 
+        {
+            std::vector<bp2::pid_type> tmp1;
+            std::vector<bp2::pid_type> tmp2 = bp2::child_pids(children[i]);
+            tmp1.insert(std::end(tmp1), std::begin(tmp2), std::end(tmp2));
+            grand_children = tmp1;
+        }
+        return (!children.empty() || !grand_children.empty());
+    };
+    #if defined(_WIN32)
+    // pid of zero is guaranteed to exist on Windows
```

Comment:
this should be unified!

Reply:
Could be that you have a `root_pid` constant that is either 1 or 0 depending on the os.

Reply:
Yeah, that's a good point. I'll create a root_pid() function then.